tutorials/text/deduplication/semantic/semantic_step_by_step.ipynb

```diff
@@ -0,0 +1,1157 @@
+    ")\n",
+    "\n",
+    "identify_duplicates_out = identify_duplicates_pipeline.run()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b47ce381",
+   "metadata": {},
+   "source": [
+    "#### Looking at Duplicates\n",
+    "\n",
+    "- `id` : This is a list of all IDs that are above our similarity threshold `eps`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "id": "865a9273",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/html": [
+       "<div>\n",
+       "<style scoped>\n",
+       "    .dataframe tbody tr th:only-of-type {\n",
+       "        vertical-align: middle;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe tbody tr th {\n",
+       "        vertical-align: top;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe thead th {\n",
+       "        text-align: right;\n",
+       "    }\n",
+       "</style>\n",
+       "<table border=\"1\" class=\"dataframe\">\n",
+       "  <thead>\n",
+       "    <tr style=\"text-align: right;\">\n",
+       "      <th></th>\n",
+       "      <th>id</th>\n",
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <th>0</th>\n",
+       "      <td>115</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>1</th>\n",
+       "      <td>268</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>2</th>\n",
+       "      <td>273</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>3</th>\n",
+       "      <td>280</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>4</th>\n",
+       "      <td>323</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "    id\n",
+       "0  115\n",
+       "1  268\n",
+       "2  273\n",
+       "3  280\n",
+       "4  323"
+      ]
+     },
+     "execution_count": 14,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "pd.read_parquet(os.path.join(duplicates_output_path, os.listdir(duplicates_output_path)[0])).head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "88140f70",
+   "metadata": {},
+   "source": [
+    "## Removing Duplicates\n",
+    "\n",
+    "We offer a simple `TextDuplicatesRemovalWorkflow` that can remove duplicates from a given input dataset and list of duplicates to remove. \n",
+    "\n",
+    "### Notes\n",
+    "1. When running the removal workflow, we must specify the same input configuration as we did when we \"generated IDs\".\n",
+    "2. In this tutorial that happened at the embedding generation step.\n",
+    "3. Therefore it's required that we match the same arguments of filepath, filetype and `files_per_partition`/`blocksize`.\n",
+    "4. This is required because IDs are generated based on hash(filenames) in each task. If the same hash is not found in the ID Generator it'll error out.\n",
```

Comment:
Nit: the meaning for "hash(filenames)" is still not immediately obvious to me, could it be written another way?